### PR TITLE
feat: Added `lint` command to validate extension source

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/web-ext/issues/204

### CODE REVIEW

File: src/cmd/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /* @flow */
 import build from './build';
 import run from './run';
+import lint from './lint';
 import sign from './sign';
-export default {build, run, sign};
+export default {build, run, sign, lint};
```
Comment (muffinresearch) [inline src/cmd/index.js]: Looks like these are sorted with the exception of the addition.

File: src/program.js [hunk 2]
```diff
@@ -243,6 +243,37 @@ Example: $0 --help run.
...
+        alias: 'o',
+        describe: 'The type of output to generate',
+        type: 'string',
+        default: 'text',
+        choices: ['json', 'text'],
+      },
+      'metadata': {
+        describe: 'Output only metadata as JSON',
+        type: 'boolean',
+        default: 'false',
```
Comment (muffinresearch) [inline src/program.js]: Shouldn't this be `false` not `'false'`

**Comment by muffinresearch** [comment]
I wonder if there would be a way to import the yargs config from the linter directly so you can expose the linter as a command and get all the cli features in the linter? Then as the linter changes you should just get the added functionality and it would help avoid any potential breakage should something in the linter change.

If this is working for now then it makes sense to not block on it, however, if there's anything that can be done on the linter side to make integration better/easier in future feel free to add issues there.

r+wc

File: src/program.js (hunk 2 above)
Comment (kumar303) [inline src/program.js]: oh, heh, copy/paste error. Thanks.

**Comment by kumar303** [comment]
> I wonder if there would be a way to import the yargs config from the linter directly

I thought about this too. The upside, as you mentioned, is that new linter program options would magically appear in web-ext. The downside is that it would be hard to know if any of those magic additions would make sense for web-ext or not. For example, it might break web-ext or a new option may be something that web-ext should handle directly (such as how it handles `sourceDir`).

So, yeah, I'm not entirely sure. Copying and pasting the configuration into web-ext makes it very explicit which is nice. If a new option gets added to `linter` I envision someone filing a bug in web-ext saying "you don't support X." This doesn't seem so bad.

**Pull request merged**